boards/BMS/Test/Src/imd_test.cpp

```diff
@@ -0,0 +1,267 @@
+#include "fff.h"
+#include <gtest/gtest.h>
+#include <utility>
```

Comment:
nit; this should be `Imd_test.cpp` to be consistent with the naming of our other files.

Reply:
Yep I can do that. However, right now the names are kinda all over  the place
```
./DIM/Test/Src/seven_seg_displays_test.cpp
./DIM/Test/Src/state_machine_test.cpp
./PDM/Test/Src/voltage_monitor_tests.cpp
./PDM/Test/Src/state_machine_test.cpp
./FSM/Test/Src/state_machine_test.cpp
./DCM/Test/Src/brake_light_tests.cpp
./DCM/Test/Src/state_machine_test.cpp
./BMS/Test/Src/imd_test.cpp
./BMS/Test/Src/state_machine_test.cpp
```

Might be worth it to clean up everything in one pass after this PR as Test_{ModuleName}.cpp?

That way we have `{App|Io}_{ModuleName}.c` for the application and IO layer code, and `Test_{ModuleName}.cpp` for all test files?

Reply:
Seems reasonable to me.

Reply:
Ok cool I would prefer to do the clean up after this PR then since it's getting a bit long :+1:

Reply:
I changed my mind and decided to just clean up all the test files in this PR anyway.